src/index/VocabularyGenerator.cpp

```diff
@@ -22,100 +22,78 @@ class PairCompare {
 };
 // ___________________________________________________________________
 void mergeVocabulary(const std::string& basename, size_t numFiles) {
-  const size_t bufferSize = 5; //see comment below for buffers
   
   std::vector<std::fstream> infiles;
   // currently only one word is buffered, no matter how big the bufferSize is
   // TODO: buffer, or throw the buffer out
-  std::vector<std::vector<std::string>> mergeBuf(numFiles);
   std::ofstream outfile(basename + ".vocabulary");
   AD_CHECK(outfile.is_open());
   std::ofstream outfileExternal(basename + EXTERNAL_LITS_TEXT_FILE_NAME);
   AD_CHECK(outfileExternal.is_open());
   std::vector<bool> endOfFile(numFiles, false);
-  std::vector<std::streampos> posNextWord;
-  std::vector<std::vector<std::string>::iterator> iterators;
+
+  using pair_T = std::pair<std::string, size_t>;
+  std::priority_queue<pair_T, std::vector<pair_T>, PairCompare> queue;
```

Comment:
I think this is not possible, since I need the third template parameter (the custom comparator) and one cannot "skip" template parameters to my knowledge.